### ISSUE TITLE
Warn when any invalid option is used for `local` for `dash` shell

### ISSUE THREAD

**Issue by EmilySeville7cfg** [koalaman/shellcheck#2818]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env dash

f() {
    local -r x=3
}

f
```

#### Here's what shellcheck currently says:

Nothing about -r or any other flag.

#### Here's what I wanted or expected to see:

In dash, local doesn't accept any flags.
